epsagon/wrappers/aiohttp.py

```diff
@@ -0,0 +1,61 @@
+"""
+Middleware for Python aiohttp.
+"""
+
+from __future__ import absolute_import
+import time
+import warnings
+from aiohttp.web import middleware
+
+import epsagon.trace
+import epsagon.triggers.http
+from epsagon.runners.aiohttp import AiohttpRunner
+from epsagon.common import EpsagonWarning
+from epsagon.utils import (
+    collect_container_metadata,
+    get_traceback_data_from_exception
+)
+from ..http_filters import ignore_request
+
+
+@middleware
+async def AiohttpMiddleware(request, handler):
+    """
+    aiohttp middleware to create a runner with event details
+    :param request: incoming request data
+    :param handler: original handler
+    :return: response data from the handler
+    """
+
+    if ignore_request('', request.path.lower()):
+        return await handler(request)
+
+    trace = epsagon.trace.trace_factory.get_or_create_trace()
+    trace.prepare()
+    runner = None
+    response = None
+
+    try:
+        body = await request.text()
```

Comment:
1. When happens if another client is initialized during this await? or await handler(request)?
I think it might cause another tracer to be initialized, so the current tracer will be overriden. Maybe using the trace object itself will solve that issue, not counting on the trace factory (multile_traces mode is not good here as its all running on the same thread)
2. Should this feature support aiohttp server? if so, we might do want multiple traces support

Reply:
The middleware is already co-routined async-wise, so it doesn't reach the same tracer together. That's also the reason to wrap the @middleware decorator

Reply:
not sure I get why - the trace factory is using a single tracer (default mode), which the middleware doesn't know about
In flask & django, this state ("multiple traces" by thread ID) is enabled by our instrumentation automatically

Reply:
You're right, I switched to multiple traces mode

Reply:
Fixed